source/text/Format.cpp

```diff
@@ -117,7 +117,18 @@ std::string Format::PlayTime(double timeVal)
 	return result;
 }
 
-
+//
+string Format::SecondaryAmmo(int value)
+{
+		if(value >= 10000)
+		{
+			return to_string(value / 1000) + "." + to_string((value - (value / 1000) * 1000) / 100) + "k";
+		}
+		else
+		{
+			return to_string(value);
+		}
+}
```

Comment:
```suggestion


string Format::SecondaryAmmo(int value)
{
		return value >= 10000 ? to_string(value / 1000) + "." + to_string((value - (value / 1000) * 1000) / 100) + "k" : to_string(value);
}


```
3 Lines after and before the function, also let's do this with a ternary operator.

`to_string((value - (value / 1000) * 1000) / 100` that feels like it could be improved somewhat, maybe you got some parenthesises wrong?
`value - (value / 1000) * 1000` this means value - value or am I missing smth here?

Reply:
Perhaps look to how the Credits function does its abbreviation then emulate that. If the two functions become similar enough then we could break out the guts into a helper function that just abbreviates to a given threshold.

Reply:
Could you divide, cast to int, then divide again to get the desired result? Like, divide by 100, cast to int, divide by 10, to get 1255 to 1.2k (1255 -> 12.55 -> 12 -> 1.2)

Reply:
@Ferociousfeind I tried something similar at first, but the problem is that while the result is numerically correct, since the end result is a float it will be displayed with a whole bunch of trailing zeros, and then more code has to be included to get it to display without those trailing zeros, which is also not ideal.

Reply:
@Amazinite I like this idea. The two things that would need to be differentiated would be the threshold as you mentioned, and then also the number of decimal places to display. Both of these should be able to be passed as arguments with relative ease, so I will look into it.